### ISSUE TITLE
Allow termux-share to share links

### ISSUE THREAD

**Issue by martmists** [termux/termux-packages#755]

**Comment by Neo-Oli** [comment]
There's `termux-open-url`. Does that not work for you?

**Comment by martmists** [comment]
Nope. Using latest termux, latest termux-api (both package and app) on OnePlus 3t

**Comment by fornwall** [comment]
I think it's a good idea to consolidate both `termux-open-url` and `termux-share` into a single command, perhaps named `termux-open`, that should also be moved from termux-api to the termux-tools package so that it comes preinstalled (with support for it added to the core Termux app, so Termux:API is not needed).

It's such a core useful thing to have so it would be nice to have it out of the box, and then also `xdg-open` could be added as a simple wrapper around it, which is nice for compatibility.

**Comment by fornwall** [comment]
There is now a built-in `xdg-open` command available which accepts both files and URL:s.

New users of Termux will have it installed by default with not setup needed, which is nice. Existing users need to update to Termux v0.47 and run `apt update && apt upgrade` to get the new tool!

**Comment by martmists** [comment]
As of right now `xdg-open` seems to crash when given an URL, and nothing happens when a valid filename is passed.

**Comment by vishalbiswas** [comment]
Did you update the Termux app to 0.47?

**Comment by martmists** [comment]
I updated the app, did `apt update` and `apt upgrade`. As I said, the command works fine with the `--send` flag, but `--view` seems to be broken.

**Comment by fornwall** [comment]
@martmists Which url are you using? Could you run `logcat` after a crash and paste the stack trace?

**Comment by martmists** [comment]
What I've tried:

```bash
$ termux-open "https://google.com"  # crash
$ termux-open https://google.com  # crash
$ termux-open --send "https://google.com"  # crash
$ termux-open --view "https://google.com"  # crash

$ termux-open --view some.file  # nothing happens
$ termux-open --send some.file  # The correct dialog pops up
```

Addubg a stacktrace in a second

**Comment by fornwall** [comment]
@martmists Thanks! This has been fixed in a newly released update 0.48 of the app, which should be released shortly.

The crash affected all Android versions earlier than 7.0 when opening an URL with xdg/termux-open.
